video/out/wayland_common.c

```diff
@@ -1270,6 +1273,25 @@ static struct vo_wayland_output *find_output(struct vo_wayland_state *wl, int in
     return NULL;
 }
 
+static void greatest_common_divisor(struct vo_wayland_state *wl, int a, int b) {
+    // euclidean algorithm
+    int larger;
+    int smaller;
+    if (a > b) {
+        larger = a;
+        smaller = b;
+    } else {
+        larger = b;
+        smaller = a;
+    }
+    int remainder = larger - smaller * floor(larger/smaller);
```

Comment:
This is where the crash happens, but together with #7458 it works fine.

Reply:
Yeah that was the plan. This apparently makes #7426 worse though so I'll hold off on merging this for now. The other one is good to go though.